planner/scan.go

```diff
@@ -60,17 +63,84 @@ func (n *scanNode) Kind() string {
 
 func (n *scanNode) Init() error {
 	// init the fetcher
-	if err := n.fetcher.Init(&n.desc, n.fields, n.reverse, n.showDeleted); err != nil {
+	if err := n.fetcher.Init(&n.desc, n.fields, n.filter, n.slct.DocumentMapping, n.reverse, n.showDeleted); err != nil {
 		return err
 	}
 	return n.initScan()
 }
 
 func (n *scanNode) initCollection(desc client.CollectionDescription) error {
 	n.desc = desc
+	n.initFields(n.slct.Fields)
+	if n.slct.OrderBy != nil {
```

Comment:
question: I think this might have been what threw me at first, and I'm still not sure why you are inspecting order conditions here. Why is this necessary, and are you really sure it is?  It does look like you could delete this block.

Reply:
This one may be changed since it was defined. I'll double check if this one is duplicating efforts, as a lot of the order by changes came after this bit. Will confirm

Reply:
Mentioned in another comment, but posting here for completeness. The `order` check/loop is now removed here.

Reply:
Nice - thanks John